### PR TITLE
Added option to disable archive checksums

### PR DESCRIPTION
**Pull request by rawkode**
Signed-off-by: David McKay <david@rawkode.com>

### CODE REVIEW

File: src/Composer/Satis/Command/BuildCommand.php [hunk 1]
```diff
@@ -414,7 +416,11 @@ private function dumpDownloads(array $config, array &$packages, InputInterface
                 $distUrl = sprintf('%s/%s/%s', $endpoint, $config['archive']['directory'], $archive);
                 $package->setDistType($archiveFormat);
                 $package->setDistUrl($distUrl);
-                $package->setDistSha1Checksum(hash_file('sha1', $path));
+
+                if ($includeArchiveChecksum === true) {
+                  $package->setDistSha1Checksum(hash_file('sha1', $path));
```
Comment (stof) [inline src/Composer/Satis/Command/BuildCommand.php]: wrong indentation

File: src/Composer/Satis/Command/BuildCommand.php [hunk 2]
```diff
@@ -414,7 +416,11 @@ private function dumpDownloads(array $config, array &$packages, InputInterface
                 $distUrl = sprintf('%s/%s/%s', $endpoint, $config['archive']['directory'], $archive);
                 $package->setDistType($archiveFormat);
                 $package->setDistUrl($distUrl);
-                $package->setDistSha1Checksum(hash_file('sha1', $path));
+
+                if ($includeArchiveChecksum === true) {
```
Comment (stof) [inline src/Composer/Satis/Command/BuildCommand.php]: ``if ($includeArchiveChecksum)`` is enough. It is already a boolean.

File: src/Composer/Satis/Command/BuildCommand.php (hunk 1 above)
Comment (rawkode) [inline src/Composer/Satis/Command/BuildCommand.php]: Fixed

**Comment by stof** [comment]
:+1:

**Comment by andrerom** [comment]
Any related doc change for this?

**Comment by alcohol** [comment]
Docs are in the [composer/composer](/composer/composer) repository. But, in short, no. Satis does not have much documentation to begin with. Feel free to contribute.

**Pull request merged**